### ISSUE TITLE
Myr Battlesphere's trigger deals damage when attacking a Battle

### ISSUE THREAD

**Issue by Zerrisx** [magefree/mage#11229]
Myr Battlesphere's trigger says it deals damage to the "Player or Planeswalker it is attacking".  A Battle is not a player or planeswalker, so no damage should be dealt, according to the MTGJudge chat.  I'm not aware of an official ruling on this, and would not be surprised to see it eventually eratta'd - but for now I believe the XMage implementation to be incorrect.

![image](https://github.com/magefree/mage/assets/59861365/1953a815-3998-486a-9b89-e0cd66b5d9bf)

**Comment by Zerrisx** [comment]
Updated the bug title, as this appears to affect multiple cards - I confirmed Hellrider and Fathom Fleet Swordjack, but I suspect it's all eight of these cards:

![image](https://github.com/magefree/mage/assets/59861365/3305eb46-77c4-45d7-b23e-77ad063cabfb)

**Comment by Zerrisx** [comment]
These three cards are also _similar_, and might or might not be affected:
![image](https://github.com/magefree/mage/assets/59861365/140c4175-fcf3-487c-9fc8-16e4352d2908)

**Comment by alexander-novo** [comment]
I really dislike the wording on these cards post-Battles. It seems to assert that there is a player or planeswalker being attacked

**Comment by Zerrisx** [comment]
I agree, and would not be surprised to see this category of cards eventually get errata similar to when damage changed from "creature or player" to "any target".

**Comment by xenohedron** [comment]
A quick look indicates that these cards are implemented with different custom effects and could use some refactoring to a common class in the process, which would also make it easier if they do get errata.
